lib/ansible/modules/network/nxos/nxos_facts.py

```diff
@@ -546,6 +550,13 @@ def main():
     runable_subsets = set()
     exclude_subsets = set()
 
+    capabilities = get_capabilities(module)
+    if capabilities:
+        os_version = capabilities['device_info']['network_os_version']
+        os_version_major = int(os_version[0])
+        if os_version_major < 7:
```

Comment:
@trishnaguha We need to support the `nxos_facts` module on the following versions.

1. 7.x and later
2. 6.0(2)A8(1) and later

You could probably just check for (> 7) and `6.0(2)A8`

Reply:
@mikewiebe Thanks. This PR is moved to https://github.com/ansible/ansible/pull/42089. Please review https://github.com/ansible/ansible/pull/42089/files#diff-83a6bf7294d0c5ac59ee458de130f795R549 and let me know if that covers all the cases.